receiver/googlecloudmonitoringreceiver/config.go

```diff
@@ -24,7 +24,8 @@ type Config struct {
 }
 
 type MetricConfig struct {
-	MetricName string `mapstructure:"metric_name"`
+	MetricName      string `mapstructure:"metric_name"`
+	MetricNameRegex string `mapstructure:"metric_name_regex"`
```

Comment:
one alternative would be to just accept the complete monitoring filter.  That way, we have metric_name for convenience, but otherwise people can do whatever they want.  That way we don't need to support every possible filter function in this configuration.  maybe `monitoring_filter`?  We can just link to the filter docs then.

WDYT?

Reply:
yeah, that will be very flexible.